### PR TITLE
Fix:local translate

### PR DESCRIPTION
**Pull request by cptbtptpbcptdtptp**
We can solve it in two ways:

1.  **modify `_position`**

```
      const distance = translation.length();
      if (distance <= MathUtil.zeroTolerance) {
        // The distance moved is too short.
        return;
      }
      const axisLen = translation.transformToVec3(this.localMatrix).length();
      if (axisLen <= MathUtil.zeroTolerance) {
        // Scale value is too extreme.
        return;
      }
      this._position.add(translation.scale(distance / axisLen));
```

2.  **modify `_worldPosition`**

```
      const distance = translation.length();
      if (distance <= MathUtil.zeroTolerance) {
        // The distance moved is too short.
        return;
      }
      const axisLen = translation.transformToVec3(this.worldMatrix).length();
      if (axisLen <= MathUtil.zeroTolerance) {
        // Scale value is too extreme.
        return;
      }
      this._worldPosition.add(translation.scale(distance / axisLen));
```

**I think modifying `_position` performs better.**

### CODE REVIEW

File: packages/core/src/Transform.ts [hunk 1]
```diff
@@ -719,7 +719,17 @@ export class Transform extends Component {
 
   private _translate(translation: Vector3, relativeToLocal: boolean = true): void {
```
Comment (GuoLei1990) [inline packages/core/src/Transform.ts:720]: Update `translate` comments with "Translate by x along the x axis, y along the y axis, and z along the z axis."

Comment (GuoLei1990) [inline packages/core/src/Transform.ts:720]: `relativeToLocal ` default value can delete

File: packages/core/src/Transform.ts [hunk 2]
```diff
@@ -717,9 +717,17 @@ export class Transform extends Component {
...
 
-  private _translate(translation: Vector3, relativeToLocal: boolean = true): void {
+  private _translate(translation: Vector3, relativeToLocal: boolean): void {
     if (relativeToLocal) {
-      this._position.add(translation);
+      const distance = translation.length();
+      if (distance <= MathUtil.zeroTolerance) {
+        return;
+      }
+      const axisLen = translation.transformToVec3(this.localMatrix).length();
```
Comment (GuoLei1990) [inline packages/core/src/Transform.ts:726]: I think this is not `localMatrix` and also not `worldMatrix`, it is `worldMatrix*invert(localMatrix)`

File: packages/core/src/Transform.ts [hunk 3]
```diff
@@ -717,9 +717,17 @@ export class Transform extends Component {
...
+      if (axisLen <= MathUtil.zeroTolerance) {
+        return;
+      }
+      this._position.add(translation.scale(distance / axisLen));
```
Comment (GuoLei1990) [inline packages/core/src/Transform.ts:730]: Here shouldn't use a uniform scale

**Comment by GuoLei1990** [comment]
You can try like this:
```ts
private _translate(translation: Vector3, relativeToLocal: boolean): void {
    if (relativeToLocal) {
      const parent = this._getParentTransform();
      if (parent) {
        const rotatedTranslation = Transform._tempVec30;
        Vector3.transformByQuat(translation, this.worldRotationQuaternion, rotatedTranslation);
        translation = rotatedTranslation;
      }
    }
    this._worldPosition.add(translation);
  }
```

**Comment by GuoLei1990** [comment]
private _translate(translation: Vector3, relativeToLocal: boolean): void {
    if (relativeToLocal) {
      const parent = this._getParentTransform();
      if (parent) {
        const rotatedTranslation = Transform._tempVec30;
        Vector3.transformByQuat(translation, this.worldRotationQuaternion, rotatedTranslation);
        translation = rotatedTranslation;
      }
    }
    this._worldPosition.add(translation);
  }

File: packages/core/src/Transform.ts (hunk 1 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/Transform.ts:720]: Done.

Comment (cptbtptpbcptdtptp) [reply packages/core/src/Transform.ts]: Done.

File: packages/core/src/Transform.ts (hunk 2 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/Transform.ts]: Fixed.

File: packages/core/src/Transform.ts (hunk 3 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/Transform.ts]: Fixed.

**Comment by cptbtptpbcptdtptp** [comment]
> 
The value of translation should not be modified.

**Pull request merged**